tools/concat-gz-files.cwl

```diff
@@ -0,0 +1,40 @@
+cwlVersion: v1.0
+class: CommandLineTool
+requirements:
+  - class: ShellCommandRequirement
+  - class: InlineJavascriptRequirement
+baseCommand: zcat
+inputs:
+  files:
+    type: File[]
+    inputBinding:
+      position: 1
+      shellQuote: False
+  pipe:
```

Comment:
With the exception of `files` and `output_filename`, the rest of these inputs should probably be  [CommandLineTool arguments](https://www.commonwl.org/v1.0/CommandLineTool.html#CommandLineTool) instead - _Command line bindings which are not directly associated with input parameters_.

Things like the `|` and `gzip` are part of the command line but not related to any input the user would plug in.

The [CommandLineTool spec](https://www.commonwl.org/v1.0/CommandLineTool.html#Input_binding) describes how the command-line is built up from inputs and arguments. 

Beyond that, any techniques we can use to make tools like this more concise (the equivalent shell script is trivial) are encouraged. I showed this during the meeting yesterday as a progress update, and I think the verbosity (compared to a `zcat *.gz | gzip > files.gz`) is discouraging.

If the tool makes sense as a shell script, one thing we could try is writing the shell script via https://www.commonwl.org/v1.0/CommandLineTool.html#InitialWorkDirRequirement.

Reply:
Switched to arguments but it is still pretty verbose. Will look into an inline shell script.

Reply:
I combined pipe and gzip.